src/extensions/replay/sessionrecording-utils.ts

```diff
@@ -165,27 +166,154 @@ export function truncateLargeConsoleLogs(_event: eventWithTime) {
+        ) {
+            // so at this point we know that the buffer is too large, and we have a single incremental snapshot
+            // it may be that a single item in the buffer is too large
+            // or that there are many small items of one or more types that end up being too large
+            // rrweb processes removes, then adds, then texts, the attributes,
+            // so we can split them in that order
+            const bufferedMutations = bufferedData.data as mutationData
+            const removes = sliceList(bufferedMutations.removes, sizeLimit)
+            const adds = sliceList(bufferedMutations.adds, sizeLimit)
+            const texts = sliceList(bufferedMutations.texts, sizeLimit)
+            const attributes = sliceList(bufferedMutations.attributes, sizeLimit)
+
+            // the incoming data has a single timestamp, so we need to adjust the timestamps of the split data,
+            // so we count how many children we have in total, and then we adjust the timestamp
+            // so that if there are 10 the first item is 9 milliseconds before the original timestamp
+            // and the final item has the original timestamp
+            const alteration =
+                countChildren(removes) + countChildren(adds) + countChildren(texts) + countChildren(attributes)
+            let timestampWiggleMarker = 1
+
+            return [
+                ...removes.map((remove) => ({
+                    size: estimateSize(remove),
+                    data: [
+                        {
+                            type: INCREMENTAL_SNAPSHOT_EVENT_TYPE,
+                            data: {
+                                source: MUTATION_SOURCE_TYPE,
+                                adds: [],
+                                texts: [],
+                                attributes: [],
+                                removes: remove,
+                            },
+                            timestamp: bufferedData.timestamp - alteration + timestampWiggleMarker++,
+                        },
+                    ],
+                    sessionId: buffer.sessionId,
+                    windowId: buffer.windowId,
+                })),
+                ...adds.map((add) => ({
+                    size: estimateSize(add),
+                    data: [
+                        {
+                            type: INCREMENTAL_SNAPSHOT_EVENT_TYPE,
+                            data: {
+                                source: MUTATION_SOURCE_TYPE,
+                                adds: add,
+                                texts: [],
+                                attributes: [],
+                                removes: [],
+                            },
+                            timestamp: bufferedData.timestamp - alteration + timestampWiggleMarker++,
+                        },
+                    ],
+                    sessionId: buffer.sessionId,
+                    windowId: buffer.windowId,
```

Comment:
Might be an opportunity to pull some of this into a function. Feels like we do almost the same thing 4 times

Reply:
good call here... 0.1% less bundle increase